### PR TITLE
Add tests for gutenberg_render_layout_support_flag

### PR DESCRIPTION
**Pull request by tellthemachines**
## What?

Adding some unit tests for `gutenberg_render_layout_support_flag` because it didn't have any.

* Test that correct base classnames are output for two layout types on a single-wrapper block;
* Test that base classname is added to inner wrapper on multi-wrapper block.

## Testing Instructions

Check out the branch and run `npm run test:unit:php `.

### CODE REVIEW

File: phpunit/block-supports/layout-test.php [hunk 1]
```diff
@@ -380,4 +380,88 @@ public function data_gutenberg_get_layout_style() {
...
+						'innerBlocks'  => array(),
+						'innerHTML'    => '<div class="wp-block-group"></div>',
+						'innerContent' => array(
+							'<div class="wp-block-group"></div>',
+						),
+					),
+				),
+				'expected_output' => '<div class="wp-block-group is-layout-constrained"></div>',
+			),
+			'multiple wrapper block layout with flow type' => array(
```
Comment (andrewserong) [inline phpunit/block-supports/layout-test.php:444]: Nice idea making sure the inner wrapper logic works 👍

File: phpunit/block-supports/layout-test.php [hunk 2]
```diff
@@ -380,4 +380,88 @@ public function data_gutenberg_get_layout_style() {
...
+
+	/**
+	 * Data provider for test_layout_support_flag_renders_classnames_on_wrapper.
+	 *
+	 * @return array
+	 */
+	public function data_layout_support_flag_renders_classnames_on_wrapper() {
+		return array(
+			'single wrapper block layout with flow type'   => array(
+				'args'            => array(
```
Comment (andrewserong) [inline phpunit/block-supports/layout-test.php:407]: This looks good to me, but I was wondering if there's a code style preference between placing the params in a nested `args` array or moving them one level up to be at the root of the array (and therefore become their own params on `test_layout_support_flag_renders_classnames_on_wrapper`)? So far I haven't been nesting them further in data providers, but it does look like a good way to keep the params contained 👍

In short: this was just a comment, not anything that needs changing IMO 🙂

**Review by andrewserong — APPROVED** [review]
Thanks for adding in these tests @tellthemachines, great idea testing to make sure the main container class is output correctly! 🎉

The tests are running well for me locally, I noticed that the Github action PHP tests are failing, but it's tag processor tests that are failing, so I assume this is a potential conflict between Gutenberg `trunk` and `wordpress-develop` rather than anything in this PR. Once that's resolved (separately to this PR), and this is rebased, it looks like this will be in good shape to land to me ✨

File: phpunit/block-supports/layout-test.php (hunk 2 above)
Comment (tellthemachines) [reply phpunit/block-supports/layout-test.php:407]: I copied this from the `get_layout_style` tests as it felt like a nice way to structure it! Not sure how widespread one or the other approach is.

Comment (andrewserong) [reply phpunit/block-supports/layout-test.php:407]: Ah yes, of course! Let's leave it as is, looks nice to me, too, I'll probably borrow it next time I'm writing a data provider 🙂

**Comment by andrewserong** [comment]
> I noticed that the Github action PHP tests are failing

Just to close the loop, the failing PHP tests should be resolved by: https://github.com/WordPress/gutenberg/pull/47720
